lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -693,7 +693,11 @@ public function computeChangeSet(ClassMetadata $class, $entity)
                 $orgValue = $originalData[$propName];
 
                 // skip if value haven't changed
-                if ($orgValue === $actualValue) {
+                if (is_object($orgValue) &&
+                    is_object($actualValue) &&
+                    $orgValue == $actualValue ||
+                    $orgValue === $actualValue
```

Comment:
I don't think this could evaluate to `true` if the previous line does not also evaluate to `true`. IMO this line is useless.

Reply:
The previous line will be evaluated only when `$orgValue` and `$actualValue` are objects.
Conditions may be written anotherly:
```
(is_object($orgValue) && is_object($actualValue) && $orgValue == $actualValue) || ($orgValue === $actualValue)
```

Reply:
This example would return true despite having different values. A `==` check isn't accurate for objects.

``` php
class Foo
{
    private $i;
    public function __construct(?int $i)
    {
        $this->i = $i;
    }
}

$f1 = new Foo(0);
$f2 = new Foo(null);

var_dump($f1 == $f2);
```

This would break, for example, with custom Doctrine types that are objects. I think this is a BC break and should target to master for the next mayor release, if this is considered as a feature.